### ISSUE TITLE
ICES Data Centre Data Validation review

### ISSUE THREAD

**Issue by neil-ices-dk** [ices-eg/DIG#371]
We are internally looking at all the data validation approaches that we currently use across all the data types; over time we have used SQL, bespoke apps, XML, Schematron etc. The idea being to see if we should be taking a new consolidated approach, and finding out if so, what that approach might look like from a process/technology perspective.

As part of the analysis steps, we are looking at current and best practices from the marine data management community. We initially started with an extensive survey, but after feedback from Jens, Dave and Sjur we have slimmed it down to a more basic canvassing, with the idea that we could then follow-up with specific respondents if their answers have information that we would like to expand on. To that end, the draft survey is here [Data Validation Survey](https://forms.office.com/r/y12inR3fLt)

_the timeline below is a bit behind schedule, but it is indicative of the steps_
![image](https://user-images.githubusercontent.com/23527005/192726831-bf415858-99bd-416e-9782-52d42f764207.png)

**Comment by mehdiabbasi** [comment]
The result of the Data validation survey and also new New Data Validation Framework Proposal​ will be presented at the May DIG meeting
[https://community.ices.dk/Committees/DIG/DIG%202023/05.%20Presentations/ICES%20new%20data%20validation%20framework%20proposal.pptx](https://community.ices.dk/Committees/DIG/DIG%202023/05.%20Presentations/ICES%20new%20data%20validation%20framework%20proposal.pptx)

**Comment by davidcurrie2001** [comment]
@mehdiabbasi are there any updates on this topic?  Are you planning to present something at the DIG meeting in May?

**Comment by mehdiabbasi** [comment]
@davidcurrie2001 yes, I am planning to update DIG about Data Validation Phase II progress but this will be included in the half a day DC presentation.

**Comment by mehdiabbasi** [comment]
@davidcurrie2001  We will present the progress on Data Validation framework design - Phase II in this presentation [DIG-Data validation framework-PhaseII](https://community.ices.dk/Committees/DIG/DIG%202024/05.%20Presentations/DIG-Data%20validation%20framework-PhaseII.pptx). We will also present [data Validation Requirements documents](https://community.ices.dk/Committees/DIG/DIG%202024/04.%20Working%20documents/DataValidationRequirements.docx) and would like to ask DIG to review these requirements and give us feedback on each feature, and whether there are any requirements needed by the community that are missing

**Comment by Ingeborg1312** [comment]
General comment on the report presented at DIG 2024: it may help to define 'target audience' to all paragraphs in chapter 2, to get clear who could provide feedback, or be asked as test panel/reviewers, especially when part of the target audience is outside the ICES Data Centre (for example DIG, governance groups, data submitters)

**Comment by Ingeborg1312** [comment]
General comment on the report: for improved overview is may be worth to organise the different bullets in a topical manner per paragraph. Eg. in 2.1.4 the first, second and sixth bullet are about validation reports, and the other about formats.

**Comment by davidcurrie2001** [comment]
Related to #524, #551, #572.

**Comment by davidcurrie2001** [comment]
Update at May 2025 meeting.

The ICES Data Validation Framework (DVF) supports a range of input formats, including CSV, JSON, XML, and more. It produces detailed validation reports highlighting errors and warnings. The solution employs modern, cloud-native technologies to deliver high performance, modular validation logic decoupled from underlying databases, and distributed deployment on Azure.

Phase II of the project focused on analysis and prototyping, delivering key outputs such as a requirements specification, sample datasets, a domain model, and a GUI prototype for user interaction.

Phase III, now in progress, is centred on developing DVF Version 1.0. Core features include:
•	Support for rule and schema definitions using JSONPath, XPath, SQL, and API-based validation
•	Template and rule reusability
•	Dataset versioning
•	Distributed, scalable processing on Azure

Development of Phase III began in June 2025, with a beta release expected by February 2026, and full production deployment planned for the end of 2026.

**Comment by mehdiabbasi** [comment]
Update at May 2026 meeting. The roadmap reflects a phased and forward-looking strategy, with Version 1.0 establishing the foundational platform and Version 2.0 introducing more advanced capabilities such as dataset versioning, automated comparisons, and AI-assisted validation. The planned migration from legacy systems further consolidates validation processes into a single framework. Overall, DVF represents a strategic investment in data governance and quality assurance, strengthening ICES’s ability to manage complex data pipelines and support high-quality, trusted data products. 

<img width="1248" height="663" alt="Image" src="https://github.com/user-attachments/assets/33fd35ee-ee13-4801-8389-293dcc266000" />  

<img width="1253" height="658" alt="Image" src="https://github.com/user-attachments/assets/63d50f5c-8a0f-443c-8a61-9e3ddbfc3c94" />
